### PR TITLE
[Frontend] Collab Page

### PR DESCRIPTION
**Pull request by kenvynKwek**
## changes:
- frontend design for collab page
- question information is hardcoded for now (need to set up restful api after matching page)
- qx's realtime textarea service tested after changes and still works

image

## qx realtime textarea service (big ups):
- instructions to run
- cd /backend/collab_service => "node server.js"
  - ensure firebaseCredientials.json are in /config
  - ensure express packaged is installed ("npm install express at /collab_service dir)
- new vscode terminal => cd /frontend => npm run dev
- open browser => localhost:5173/collab

https://github.com/user-attachments/assets/1ccb3e31-ef00-4ffe-aaaf-f9b1d139f6f8

<br />  
planning to look into chat feature next

### CODE REVIEW

**Comment by kenvynKwek** [comment]
## removed question id from collab-page
image

removed question id from collab-page as per feedback in telegram as the stored question id will be a string of random characters

File: backend/collab_service/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# PeerPrep Backend
+
+## Project Overview
+The PeerPrep backend provides a service for managing and querying questions used for technical interview practice. It supports creating, retrieving, and deleting questions.
```
Comment (Tan-Jia-Rong) [inline backend/collab_service/README.md]: Update this line to what the collab service does

File: backend/collab_service/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+   ```sh
+   cd path/to/backend/
+   docker build -t peerprep-backend .
+   ```
+
+3. **Create and Run the Docker Container**
+   
+   ```sh
+   docker run -d -p 5001:5001 --name peerprep-backend-app peerprep-backend
+   ```
```
Comment (Tan-Jia-Rong) [inline backend/collab_service/README.md]: This part too

File: backend/collab_service/server.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    cors: {
+      origin: ['http://localhost:3000', 'http://localhost:5173'], // Frontend origins
+      methods: ['GET', 'POST'],
+      credentials: true // Allow cookies/credentials if needed
+    }
+  });
+
+const collabRoutes = require('./routes/collabRoutes');
+app.use(express.json());
+app.use('/', collabRoutes);
```
Comment (Tan-Jia-Rong) [inline backend/collab_service/server.js]: Not needed already defined in app.js

File: backend/collab_service/testserver.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+const io = require('socket.io-client');
+
+// Connect to the Socket.IO server
+const socket = io('http://localhost:5001', {
```
Comment (Tan-Jia-Rong) [inline backend/collab_service/testserver.js]: socket is at 5004

File: backend/collab_service/testserver.js
Comment (Tan-Jia-Rong) [inline backend/collab_service/testserver.js:1]: Assuming this file is deleted later or smthing?

**Review by Tan-Jia-Rong — APPROVED** [review]
Backend a few comments but frontend looks good. Feel free to merge in!

File: backend/collab_service/README.md (hunk 1 above)
Comment (kenvynKwek) [reply backend/collab_service/README.md]: README.md updated, thanks !

File: backend/collab_service/server.js (hunk 3 above)
Comment (kenvynKwek) [reply backend/collab_service/server.js]: Got it 👌 these 3 lines removed & double checked route still works

File: backend/collab_service/testserver.js (hunk 4 above)
Comment (kenvynKwek) [reply backend/collab_service/testserver.js]: `testserver.js` removed, no longer needed

**Pull request merged**